web/modules/custom/eulogin/src/Event/Subscriber/JoinupEuLoginKernelSubscriber.php

```diff
@@ -206,7 +209,9 @@ protected function userHasUnlimitedAccess(): bool {
     // - is anonymous.
     return $this->currentUser->isAnonymous()
     // - or is an EU Login linked user.
-    || $this->authmap->get($this->currentUser->id(), 'cas');
+    || $this->authmap->get($this->currentUser->id(), 'cas')
+    // - or is granted with 'unlimited access' permission.
+    || $this->currentUser->hasPermission('unlimited access');
```

Comment:
We are checking a role, a state and a permission. We can already avoid one by giving the permission to the anonymous user now.

Also this could use some more explanation, for me as a reviewer not familiar with this code base I am unsure what this means that the user has "unlimited access". It sounds very drastic, as if they get access rights similar to UID1, but it is probably just a naming issue.

Reply:
I reconsider the suggestion to assign this permission to anonymous users. I think it is better to check it explicitly like is done now.